buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KotlinNativeTest.kt

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+package org.jetbrains.kotlin
+
+import groovy.lang.Closure
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.TaskAction
+import org.gradle.language.base.plugins.LifecycleBasePlugin
+import org.gradle.process.ExecSpec
+
+import java.io.File
+import java.io.ByteArrayOutputStream
+import java.util.regex.Pattern
+
+import org.jetbrains.kotlin.konan.target.HostManager
+
+abstract class KonanTest : DefaultTask() {
+    enum class Logger {
+        EMPTY,    // Built without test runner
+        GTEST,    // Google test log output
+        TEAMCITY, // TeamCity log output
+        SIMPLE,   // Prints simple messages of passed/failed tests
+        SILENT    // Prints no log of passed/failed tests
+    }
+
+    var disabled: Boolean
+        get() = !enabled
+        @Optional set(value) { enabled = !value }
+
+    /**
+     * Test output directory. Used to store processed sources and binary artifacts.
+     */
+    lateinit var outputDirectory: String
+
+    /**
+     * Test logger to be used for the test built with TestRunner (`-tr` option).
+     */
+    @Optional
+    lateinit var testLogger: Logger
+
+    /**
+     * Test executable arguments.
+     */
+    @Input
+    lateinit var arguments: MutableList<String>
+
+    /**
+     * Test executable.
+     */
+    @Input
+    lateinit var executable: String
+
+    /**
+     * Test source.
+     */
+    @Optional
+    lateinit var source: String
+
+    /**
+     * Sets test filtering to choose the exact test in the executable built with TestRunner.
+     */
+    @Input
+    var useFilter = true
+
+    /**
+     * An action to be executed before the build.
+     * As this run task comes after the build task all actions for doFirst
+     * should be done before the build and not run.
+     */
+    @Optional
+    var doBefore: Action<in Task>? = null
+
+    @Suppress("UnstableApiUsage")
+    override fun configure(config: Closure<*>): Task {
+        super.configure(config)
+
+        // Set Gradle properties for the better navigation
+        group = LifecycleBasePlugin.VERIFICATION_GROUP
+        description = "Kotlin/Native test infrastructure task"
+
+        if (!::arguments.isInitialized) {
```

Comment:
Can we just initilize `arguments` with an empty list and get rid of `lateinit` and this check?